### ISSUE TITLE
Manage multiple charts of account.

### ISSUE THREAD

**Issue by danelowe** [crftr/double_double#3]
The project that I'm working on at the moment requires transactions in multiple currencies. 

I'd made some changes to make my tests pass when using a currency other than Money.default_currency (https://github.com/danelowe/double_double/commit/99ab2fd3c4734a4aaf269640c6eab11b41372ee9), but then as I was committing I realised that the solution would actually be to create a form of multi-tenancy, where accounts could (but wouldn't have to) belong to a chart of accounts. 

Each chart of accounts could then have their own base currency, and all amounts in any of it's accounts would be forced to use that currency.

What are your thoughts on this? I'll get started on it as soon as I come to that stage in the project.

**Comment by danelowe** [comment]
would it be ChartOfAccounts, or AccountSet, or?

**Comment by crftr** [comment]
I see three potential features here.

A.  Supporting multi tenancy.
B.  Accounts of mixed currencies.
C.  Enforcing a base currency.

**If multi tenancy is the goal** (A): I agree that creating an abstraction of a new ChartOfAccounts model and then assigning accounts to a particular COA would be proper.  If your project is a SAAS offering and there's the potential for users to define their own accounts, sandboxing each user to their own COA would be smart.

**If multi currency support is the goal** (B): stay the course with your revision (danelowe@99ab2fd).  It's a solid path.  If your project just needs to support multiple currencies, it really should exist in the same Chart.

Thinking further, I also like your idea of a forced base currency (C), but store that attribute on the Account.  I would advocate that it's up to the implementing developer to choose how much restriction to place on their accounts.

*Where the account is forced to a specific currency if the currency is defined on the account.  Otherwise the account is potentially a mixed currency account.*

*e.g.*  Imagine a contrived wallet app.  I would think it should be possible to have strict asset accounts, but to not require strict expense or liability accounts.

    DoubleDouble::Asset.create! name: 'Cash', number: 1, currency: :aus
    DoubleDouble::Asset.create! name: 'Bitcoins', number: 2, currency: :btc
    DoubleDouble::Expense.create! name: 'Spending', number: 3


I'm rambling a bit, but at this point I'm game for any combination of the three features, above.

Are we on the same page?

**Comment by crftr** [comment]
After sitting on this for a day-- I would gladly merge your multi currency work.  I think there's also an opportunity to write a balance method that doesn't simplify down to one currency.  

*i.e.* add a method `balance_without_conversion` that returns a hash with each respective currency amount.  `{usd: 500, aus: 800}`

But you need a ChartOfAccounts abstraction for your project, let's do it.

**Comment by danelowe** [comment]
Hi Mike, 

I have to run soon.

My project isn't actually SaaS, but will require operating accounts in a couple of different regions and currencies. At the moment I'm working on implementing the multiple Charts of Accounts options, because:

a) It's more flexible, and 
b) To the best of my understanding, it makes more sense from an accounting perspective, as I don't think multiple currencies should be used in the same chart of accounts, and it causes a number of issues, and 
c) For my use case at least, I don't see any downsides to having multiple charts of accounts. It will be easy to set up the same accounts under multiple charts. 

Basically, at this point I am probably going to work on it in order of A, C, then B. And B would have to be specifically enabled for those who don't care too much about things lining up correctly. 

I will come back later with more details, but just thought I would let you know where I am heading at the moment.

**Comment by crftr** [comment]
I hear you, Dane.  While we may not currently agree on the use-case solution, we appear to agree on the features, themselves.

I'm closing this issue, and its place, creating three new issues to track each proposed feature.
